Add JS release notes for June (#1386)
Azure/azure-sdk: releases/2020-06/js.md

The Azure SDK team is pleased to make available the June 2020 client library release.

#### GA

- _Add packages_

#### Updates

- _Add packages_

#### Preview

> - _Add packages_

Reviewer: This is the GA release of Text Analytics.

Author: @willmtemple other languages list both preview.5 and GA.  Should we do that?  Feel free to submit suggestion if needed.

### Text Analytics

This is the first GA release version of Text Analytics library.

#### Breaking Changes

- Renamed all result array types that extend JavaScript's base `Array` class to end with the word `Array` instead of `Collection` (e.g. `AnalyzeSentimentResultCollection` is now `AnalyzeSentimentResultArray`)
- Renamed `score` to `confidenceScore` in the `Match`, `Entity`, and `DetectedLanguage` types.
- Removed the `graphemeOffset` and `graphemeLength` properties of the `Match`, `Entity`, and `SentenceSentiment` types.
- Renamed the `graphemeCount` property of `TextDocumentStatistics` back to `characterCount`
- Removed the `warnings` property of `SentenceSentiment`.

#### New Features

- The library now uses the Text Analytics v3.0 (General Availability) service endpoint.
- Added a `text` property to `SentenceSentiment` that contains the sentence text
- Added `warnings` property to every document response object containing a list of `TextAnalyticsWarning` objects relevant to the corresponding document.

### Event Hubs

#### Key Bug Fixes

- Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

### Identity

#### Breaking Changes

> - TODO

Reviewer 2: There is a breaking change coming in for Identity:

- Switch to using CredentialUnavailable to differentiate from expected and unexpected errors during DefaultAzureCredential startup. [PR #8172](https://github.com/Azure/azure-sdk-for-js/pull/8127). Note: this changes the original behavior for ManagedIdentityCredential, which previously would return `null` instead of throwing an exception.

#### New Features

> - TODO

Reviewer 2: - Switch to using CredentialUnavailable to differentiate from expected and unexpected errors during DefaultAzureCredential startup. [PR #8172](https://github.com/Azure/azure-sdk-for-js/pull/8127)

- Make all developer credentials public as well as the list used by DefaultAzureCredential [PR #9274](https://github.com/Azure/azure-sdk-for-js/pull/9274)

#### New Features

- Added separate methods for getting just names such as `listIndexesNames`, `listSynonymMapsNames`, etc.
- Added `getSearchClient` method to the `SearchIndexClient` class.

### Form Recognizer

#### Breaking Changes

- Recognition and training methods now return results directly, instead of wrapping the result in a response object.
- Rename refactoring for many public APIs for cross-language consistency.
- Remove `USReceipt` and related types.

#### New Features

- Add support to copy custom model from one Form Recognizer resource to another.
- Add support for Azure Active Directory credential.

### Service Bus

#### Breaking Changes

- `ServiceBusClient.createSender()` which was made async in the previous preview to include the link initialization is no longer async. Instead, the sender now has an `open()` method that can be used to proactively initialize the link. 
- `Receiver/SessionReceiver.browseMessages()` has been renamed to `Receiver/SessionReceiver.peekMessages()`.

#### New Features

> - TODO

Reviewer 3: How about this? @richardpark-msft

```suggestion
- Added management api features under `ServiceBusManagementClient` that lets you manage a namespace with CRUD operations on the entities(queues, topics and subscriptions) and on the rules of a subscription.
	- Supports authentication with a service bus connection string as well as with the AAD credentials from `@azure/identity` along with the host name of your namespace.
	- All operations return promises that resolve to an object that has the relevant output. These objects also have a property called `_response` that you can use if you want to access the direct response from the service.
```

Reviewer 3: @ramya-rao-a

Reviewer 4: You can clip this, I think (in between --> and <--):

- Supports authentication with a service bus connection string as well as with the AAD credentials from `@azure/identity` ---> along with the hostname of your namespace. <---

And I think this is just a given so probably not needed:

> All operations return promises that resolve to an object that has the relevant output. These objects also have a property called `_response` that you can use if you want to access the direct response from the service.

The rest of it is awesome, and it's great to have the callout to identity.

Reviewer 3: Updated. Feel free to tweak it :)

### Text Analytics

This is the first GA release version of Text Analytics library.

#### Breaking Changes

- Renamed all result array types that extend JavaScript's base `Array` class to end with the word `Array` instead of `Collection` (e.g. `AnalyzeSentimentResultCollection` is now `AnalyzeSentimentResultArray`)
- Renamed `score` to `confidenceScore` in the `Match`, `Entity`, and `DetectedLanguage` types.
- Removed the `graphemeOffset` and `graphemeLength` properties of the `Match`, `Entity`, and `SentenceSentiment` types.
- Renamed the `graphemeCount` property of `TextDocumentStatistics` back to `characterCount`
- Removed the `warnings` property of `SentenceSentiment`.

#### New Features

- The library now uses the Text Analytics v3.0 (General Availability) service endpoint.
- Added a `text` property to `SentenceSentiment` that contains the sentence text
- Added `warnings` property to every document response object containing a list of `TextAnalyticsWarning` objects relevant to the corresponding document.

### Event Hubs

#### Key Bug Fixes

- Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

### Identity

> #### Breaking Changes
> 
> - TODO
> 
> #### New Features
> 
> - TODO
> 
> #### Key Bug Fixes
> 
> - TODO
> 
> - TODO

Author: ```suggestion
#### Breaking Changes

- Switch to using `CredentialUnavailable` to differentiate from expected and unexpected errors during `DefaultAzureCredential` startup. [PR #8172](https://github.com/Azure/azure-sdk-for-js/pull/8127). Note: this changes the original behavior for `ManagedIdentityCredential`, which previously would return `null` instead of throwing an exception.

#### New Features

- Make all developer credentials public as well as the list used by `DefaultAzureCredential` [PR #9274](https://github.com/Azure/azure-sdk-for-js/pull/9274)
```

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### Event Hubs

#### Key Bug Fixes

> - Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

Reviewer 5: @chradek Am thinking a generic statement on connection reliability should be good enough?

### Identity

#### Breaking Changes

- TODO

#### New Features

- TODO

#### Key Bug Fixes

- TODO

- TODO

### Search

> #### Breaking Changes

Reviewer 6: The following code has to be added here

```
#### Breaking Changes

- Changed `ListIndexes` operation to a pageable operation.
- Added `onlyIfUnchanged` parameter for CreateOrUpdate and Delete operations.
- Removed `$select` property for the List operations.
- Refactored `SearchServiceClient` and split it to `SearchIndexClient` and `SearchIndexerClient` and changed `SearchIndexClient` class to `SearchClient` class.
- Moved non-optional parameters from options bag.
- Renamed `countDocuments` method to `getDocumentsCount` method.
- In `search` method, moved the `searchText` parameter from the options bag to method parameter. 
- In `indexDocuments` method, the options parameter is renamed to `IndexDocumentsOptions`.
- Modified `deleteDocuments` method to get documents as a parameter.
- In `getIndexStatistics` method, renamed the return type from `GetIndexStatisticsResult` to `SearchIndexStatistics`.
- In `getServiceStatistics` method, renamed the return type from `ServiceStatistics` to `SearchServiceStatistics`.
- Modified `DataSource` model name to `DataSourceConnection`. Changed all references in the method names, parameters, etc.
- Renamed `SimpleDataType` model to `SearchFieldDataType` model.
- Modified the names of several models & parameters. Please refer [#8984](https://github.com/Azure/azure-sdk-for-js/issues/8984), [#9037](https://github.com/Azure/azure-sdk-for-js/issues/9037) and [#8383](https://github.com/Azure/azure-sdk-for-js/issues/8383) for a detailed list of renames.

#### New Features

- Added separate methods for getting just names such as `listIndexesNames`, `listSynonymMapsNames`, etc.
- Added `getSearchClient` method to the `SearchIndexClient` class.
```

Author: I tweak some of the wording and committed.  Please review and submit additional suggestions

- Renamed `score` to `confidenceScore` in the `Match`, `Entity`, and `DetectedLanguage` types.
- Removed the `graphemeOffset` and `graphemeLength` properties of the `Match`, `Entity`, and `SentenceSentiment` types.
- Renamed the `graphemeCount` property of `TextDocumentStatistics` back to `characterCount`
- Removed the `warnings` property of `SentenceSentiment`.

#### New Features

- The library now uses the Text Analytics v3.0 (General Availability) service endpoint.
- Added a `text` property to `SentenceSentiment` that contains the sentence text
- Added `warnings` property to every document response object containing a list of `TextAnalyticsWarning` objects relevant to the corresponding document.

### Form Recognizer

#### Breaking Changes

- Recognition and training methods now return results directly, instead of wrapping the result in a response object.
- Rename refactoring for many public APIs for cross-language consistency.
- Remove `USReceipt` and related types.

#### New Features

- Add support to copy custom model from one Form Recognizer resource to another.
- Add support for Azure Active Directory credential.

### Service Bus

#### Breaking Changes

> - TODO

Reviewer 4: ```suggestion
- `Sender` now has an `open()` method to proactively initialize the connection. In addition `ServiceBusClient.createSender()` is no longer an `async` method.
```

#### New Features

> - TODO

Reviewer 4: @HarshaNalluru  - when you're ready go ahead and document the namespace change.

> - `Sender` now has an `open()` method to proactively initialize the connection. In addition `ServiceBusClient.createSender()` is no longer an `async` method.

Reviewer 4: ```suggestion
- `ServiceBusClient.createSender()` which was made async in the previous preview to include the link initialization is no longer async. Instead, the sender now has an `open()` method that can be used to proactively initialize the link. 
```

#### Breaking Changes

- Renamed all result array types that extend JavaScript's base `Array` class to end with the word `Array` instead of `Collection` (e.g. `AnalyzeSentimentResultCollection` is now `AnalyzeSentimentResultArray`)
- Renamed `score` to `confidenceScore` in the `Match`, `Entity`, and `DetectedLanguage` types.
- Removed the `graphemeOffset` and `graphemeLength` properties of the `Match`, `Entity`, and `SentenceSentiment` types.
- Renamed the `graphemeCount` property of `TextDocumentStatistics` back to `characterCount`
- Removed the `warnings` property of `SentenceSentiment`.

#### New Features

- The library now uses the Text Analytics v3.0 (General Availability) service endpoint.
- Added a `text` property to `SentenceSentiment` that contains the sentence text
- Added `warnings` property to every document response object containing a list of `TextAnalyticsWarning` objects relevant to the corresponding document.

### Event Hubs

#### Key Bug Fixes

- Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

### Identity

#### Breaking Changes

- TODO

#### New Features

- TODO

#### Key Bug Fixes

- TODO

- TODO

### Search

> #### Breaking Changes
> 
> - TODO
> 
> #### New Features
> 
> - TODO
> 
> #### Key Bug Fixes
> 
> - TODO

Author: ```suggestion
#### Breaking Changes

- Changed `ListIndexes` operation to a pageable operation.
- Added `onlyIfUnchanged` parameter for CreateOrUpdate and Delete operations.
- Removed `$select` property for the List operations.
- Refactored `SearchServiceClient` and split it to `SearchIndexClient` and `SearchIndexerClient` and changed `SearchIndexClient` class to `SearchClient` class.
- Moved non-optional parameters from options bag.
- Renamed `countDocuments` method to `getDocumentsCount` method.
- In `search` method, moved the `searchText` parameter from the options bag to method parameter. 
- In `indexDocuments` method, the options parameter is renamed to `IndexDocumentsOptions`.
- Modified `deleteDocuments` method to get documents as a parameter.
- In `getIndexStatistics` method, renamed the return type from `GetIndexStatisticsResult` to `SearchIndexStatistics`.
- In `getServiceStatistics` method, renamed the return type from `ServiceStatistics` to `SearchServiceStatistics`.
- Modified `DataSource` model name to `DataSourceConnection`. Changed all references in the method names, parameters, etc.
- Renamed `SimpleDataType` model to `SearchFieldDataType` model.
- Modified the names of several models & parameters. Please refer [#8984](https://github.com/Azure/azure-sdk-for-js/issues/8984), [#9037](https://github.com/Azure/azure-sdk-for-js/issues/9037) and [#8383](https://github.com/Azure/azure-sdk-for-js/issues/8383) for a detailed list of renames.

#### New Features

- Added separate methods for getting just names such as `listIndexesNames`, `listSynonymMapsNames`, etc.
- Added `getSearchClient` method to the `SearchIndexClient` class.```

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### Text Analytics

This is the first GA release version of Text Analytics library.

#### Breaking Changes

- Renamed all result array types that extend JavaScript's base `Array` class to end with the word `Array` instead of `Collection` (e.g. `AnalyzeSentimentResultCollection` is now `AnalyzeSentimentResultArray`)
- Renamed `score` to `confidenceScore` in the `Match`, `Entity`, and `DetectedLanguage` types.
- Removed the `graphemeOffset` and `graphemeLength` properties of the `Match`, `Entity`, and `SentenceSentiment` types.
- Renamed the `graphemeCount` property of `TextDocumentStatistics` back to `characterCount`
- Removed the `warnings` property of `SentenceSentiment`.

#### New Features

- The library now uses the Text Analytics v3.0 (General Availability) service endpoint.
- Added a `text` property to `SentenceSentiment` that contains the sentence text
- Added `warnings` property to every document response object containing a list of `TextAnalyticsWarning` objects relevant to the corresponding document.

### Event Hubs

#### Key Bug Fixes

> - Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

Reviewer 7: @ramya-rao-a Makes a good point. How about:
```suggestion
- Improved the reliability of connections and connection recovery.
```

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

> ### Event Hubs
> 
> #### Breaking Changes
> 
> - TODO
> 
> #### New Features
> 
> - TODO
> 
> #### Key Bug Fixes
> 
> - TODO
> 

Reviewer 7: Event Hubs only has some bug fixes that improve resiliency when a connection is disconnected, and tracing updates.
Since we aren't documenting tracing updates, all we have are bug fixes.
Note: neither of the bug fixes require the user to change their code. I'm including one of the fixes here since it has been reported specifically by customers:
```suggestion
### Event Hubs

#### Key Bug Fixes

- Fixed an issue where a `TypeError` was sometimes thrown as an uncaught exception.

```

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### Text Analytics

> Text Analytics library is now GA release version.

Author: ```suggestion
This is the first GA release version of Text Analytics library.
```

- Modified `DataSource` model name to `DataSourceConnection`. Changed all references in the method names, parameters, etc.
- Renamed `SimpleDataType` model to `SearchFieldDataType` model.
- Modified the names of several models & parameters. Please refer [#8984](https://github.com/Azure/azure-sdk-for-js/issues/8984), [#9037](https://github.com/Azure/azure-sdk-for-js/issues/9037) and [#8383](https://github.com/Azure/azure-sdk-for-js/issues/8383) for a detailed list of renames.

#### New Features

- Added separate methods for getting just names such as `listIndexesNames`, `listSynonymMapsNames`, etc.
- Added `getSearchClient` method to the `SearchIndexClient` class.

### Form Recognizer

#### Breaking Changes

- Recognition and training methods now return results directly, instead of wrapping the result in a response object.
- Rename refactoring for many public APIs for cross-language consistency.
- Remove `USReceipt` and related types.

#### New Features

- Add support to copy custom model from one Form Recognizer resource to another.
- Add support for Azure Active Directory credential.

### Service Bus

#### Breaking Changes

> - `ServiceBusClient.createSender()` which was made async in the previous preview to include the link initialization is no longer async. Instead, the sender now has an `open()` method that can be used to proactively initialize the link. 

Reviewer 4: ```suggestion
- `ServiceBusClient.createSender()` which was made async in the previous preview to include the link initialization is no longer async. Instead, the sender now has an `open()` method that can be used to proactively initialize the link. 
- `Receiver/SessionReceiver.browseMessages()` has been renamed to `Receiver/SessionReceiver.peekMessages()`.
  [PR 9280](https://github.com/Azure/azure-sdk-for-js/pull/9280)
```

#### New Features

> - TODO

Reviewer 4: ```suggestion
- Added management api features which allows CRUD operations on the entities of a namespace.
```

Reviewer 4: Harsha has a better description below. Rescinding this weak version of it. :)

#### Key Bug Fixes

> - TODO

Reviewer 4: ```suggestion
- Improves the performance of the `ServiceBusMessageBatch.tryAdd` method.
  [PR 8772](https://github.com/Azure/azure-sdk-for-js/pull/8772)
```
